allow newer opentelemetry-instrumentation packages

oliverlambson opened the pull request:
the version formatting of opentelemetry-instrumentation `^0.51b0` constrains it to `>=0.51b0,<0.52` when i think the intention is actually for it to be `>=0.51b0,<1`

i also added classifiers for the newer python versions since they were missing

oliverlambson committed:
allow newer opentelemetry-instrumentation packages

the version formatting of opentelemetry-instrumentation constrains it to
an old version

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chandr-andr commented:
@oliverlambson 
Hello! Thanks for the PR.
Could you please fix the CI and then ill merge this PR.

oliverlambson commented:
@chandr-andr the errors are the same ones currently failing on master, not related to the PR:

`Executable poetry not found`

oliverlambson committed:
fix pre-commit config

oliverlambson committed:
run pre-commit

oliverlambson committed:
fix existing ruff & mypy errors

oliverlambson committed:
remove ruff ignores for rules that don't exist anymore

oliverlambson commented:
@chandr-andr I've fixed the pre-commit config and addressed the errors it was flagging

let me know if you'd prefer me to move that to a separate PR

chandr-andr commented:
Thanks! I'll merge and make new release today as soon as I come home

oliverlambson commented:
Thanks! Will you do a pypi release? Appreciate it!
